### PR TITLE
[PAR-4246] magento module display offer modal in cart

### PR DESCRIPTION
**Pull request by adityaagashe1**
# Description

Adding simpleOffer button to Magento Cart. 

## Ticket

List your ticket here

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactor (new code changes but everything functions the same)

## Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have deployed and validated my code on a sandbox
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: view/frontend/requirejs-config.js [hunk 1]
```diff
@@ -7,6 +7,8 @@ var config = {
   map: {
     '*': {
       productProtectionOffer: 'Extend_Integration/js/view/catalog/product/product-protection-offer',
+      simpleProductProtectionOffer:
+        'Extend_Integration/js/view/cart/product-protection-simple-offer',
```
Comment (scott-taylor-extend) [inline view/frontend/requirejs-config.js:11]: Can we change this not to say simple if it's going to be used for all product types?

File: view/frontend/templates/cart/item/renderer/actions/product-protection-simple-offer.phtml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    $categoryName = $category->getName();
+}
+?>
+<div>
+    <div class="product-protection-offer" id="product_protection_offer_<?= $product->getSku() ?>"></div>
+</div>
+<script type="text/x-magento-init">
+    {
+        "#product_protection_offer_<?= $product->getSku() ?>": {
+            "simpleProductProtectionOffer": [
```
Comment (scott-taylor-extend) [inline view/frontend/templates/cart/item/renderer/actions/product-protection-simple-offer.phtml:25]: Same here, lets not call it simple if its going to be used for all product types.

File: view/frontend/templates/cart/item/renderer/actions/product-protection-simple-offer.phtml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+               {
+                        "extendStoreUuid": "<?= $block
+                            ->getData('viewModel')
+                            ->getExtendStoreUuid() ?>",
+                        "activeEnvironment": "<?= $block
+                            ->getData('viewModel')
+                            ->getActiveEnvironment() ?>",
+                        "selectedProductSku": "<?= $product->getSku() ?>",
```
Comment (scott-taylor-extend) [inline view/frontend/templates/cart/item/renderer/actions/product-protection-simple-offer.phtml:33]: If you add a grouped or configurable product to the cart, is this the configurable product data, or the simple product variation?

File: view/frontend/requirejs-config.js (hunk 1 above)
Comment (jomarsantos) [reply view/frontend/requirejs-config.js:11]: I think "simple" here is in reference to Extend's idea of a "simple offer" not the Magento "simple" product type

Comment (jay-faust-extend) [reply view/frontend/requirejs-config.js:11]: I was thinking this too but I thought I was maybe ignorant on some Mage jargon.

Comment (adityaagashe1) [reply view/frontend/requirejs-config.js:11]: It is `simple` offer from Extend SDK perspective.

File: view/frontend/templates/cart/item/renderer/actions/product-protection-simple-offer.phtml (hunk 2 above)
Comment (adityaagashe1) [reply view/frontend/templates/cart/item/renderer/actions/product-protection-simple-offer.phtml:25]: This is a simple offer from Extend SDK perspective.

File: view/frontend/templates/cart/item/renderer/actions/product-protection-simple-offer.phtml (hunk 3 above)
Comment (adityaagashe1) [reply view/frontend/templates/cart/item/renderer/actions/product-protection-simple-offer.phtml:33]: Right now, we're only showing up offer for simple product.

**Review by scott-taylor-extend — DISMISSED** [review]

File: view/frontend/templates/cart/item/renderer/actions/product-protection-simple-offer.phtml (hunk 3 above)
Comment (adityaagashe1) [reply view/frontend/templates/cart/item/renderer/actions/product-protection-simple-offer.phtml:33]: It works for Grouped Product and Configurable product.

**Review by jay-faust-extend — DISMISSED** [review]

**Review by jay-faust-extend — APPROVED** [review]

**Review by jomarsantos — APPROVED** [review]

**Pull request merged**